libs/features/personalization/personalization.js

```diff
@@ -1124,6 +1128,104 @@ async function callMartech(config) {
   }
   return targetManifests;
 }
+
+function roundToQuarter(num) {
+  return Math.ceil(num / 250) / 4;
+}
+
+function calculateResponseTime(responseStart) {
+  const responseTime = Date.now() - responseStart;
+  return roundToQuarter(responseTime);
+}
+
+function sendTargetResponseAnalytics(failure, responseStart, timeoutLocal, message) {
+  // temporary solution until we can decide on a better timeout value
+  const responseTime = calculateResponseTime(responseStart);
+  const timeoutTime = roundToQuarter(timeoutLocal);
+  let val = `target response time ${responseTime}:timed out ${failure}:timeout ${timeoutTime}`;
+  if (message) val += `:${message}`;
+  // eslint-disable-next-line no-underscore-dangle
+  window._satellite?.track?.('event', {
+    documentUnloading: true,
+    xdm: {
+      eventType: 'web.webinteraction.linkClicks',
+      web: {
+        webInteraction: {
+          linkClicks: { value: 1 },
+          type: 'other',
+          name: val,
+        },
+      },
+    },
+    data: { _adobe_corpnew: { digitalData: { primaryEvent: { eventInfo: { eventName: val } } } } },
+  });
+}
+
+const handleAlloyResponse = (response) => {
+  const items = (
+    (response.propositions?.length && response.propositions)
+    || (response.decisions?.length && response.decisions)
+    || []
+  ).map((i) => i.items).flat();
+
+  if (!items?.length) return [];
+
+  return items
+    .map((item) => {
+      const content = item?.data?.content;
+      if (!content || !(content.manifestLocation || content.manifestContent)) return null;
+
+      return {
+        manifestPath: content.manifestLocation || content.manifestPath,
+        manifestUrl: content.manifestLocation,
+        manifestData: content.manifestContent?.experiences?.data || content.manifestContent?.data,
+        manifestPlaceholders: content.manifestContent?.placeholders?.data,
+        manifestInfo: content.manifestContent?.info.data,
+        name: item.meta['activity.name'],
+        variantLabel: item.meta['experience.name'] && `target-${item.meta['experience.name']}`,
+        meta: item.meta,
+      };
+    })
+    .filter(Boolean);
+};
+
+async function handleMartechTargetInteraction(
+  { config, targetInteractionPromise, calculatedTimeout },
+) {
+  let targetManifests = [];
+  let targetPropositions = [];
+  let responseStart;
+  if (enablePersonalizationV2() && targetInteractionPromise) {
+    try {
+      responseStart = Date.now();
```

Comment:
We should use `performance.mark` or `performance.now`.  `performance.now()` is a drop in replacement for `Date.now()` but `performance.mark` will generate an object containing a bit more information for you (like how long after the pageload did the measurement start).